app/src/androidTest/java/org/simple/clinic/bp/BloodPressureRepositoryAndroidTest.kt

```diff
@@ -127,6 +128,41 @@ class BloodPressureRepositoryAndroidTest {
     assertThat(savedBloodPressure).isEqualTo(expected)
   }
 
+  @Test
+  fun observing_a_deleted_blood_pressure_as_a_stream_should_work_correctly() {
+    val now = Instant.now(clock)
+    val bloodPressure = testData.bloodPressureMeasurement(
+        createdAt = now,
+        updatedAt = now,
+        deletedAt = null,
+        syncStatus = SyncStatus.DONE)
+    appDatabase.bloodPressureDao().save(listOf(bloodPressure))
+
+    val deletedMeasurementObserver = repository.deletedMeasurementAsStream(bloodPressure.uuid)
+        .test()
+
+    val fifteenSeconds = Duration.ofSeconds(15L)
+    val tenHours = Duration.ofHours(10L)
+    val twentyDays = Duration.ofDays(20L)
+
+    listOf(fifteenSeconds, tenHours, twentyDays)
+        .map { now.plus(it) }
+        .map { bloodPressure.copy(deletedAt = it) }
+        .forEach { appDatabase.bloodPressureDao().save(listOf(it)) }
+
+    // We cannot verify we received the values exactly because Room does not always emit all the
+    // intermediate changes. So we'll either have to add a delay between each update to guarantee
+    // that the change notification will be published, or just verify that the last event received
+    // is what we expect.
+    deletedMeasurementObserver.await(1L, TimeUnit.SECONDS)
```

Comment:
Alternatively, can we do a blocking get which will give us all three items?

Reply:
We can't use a `blocking` call here because the source never completes and remains alive until it is manually disposed.

Reply:
What do you think of doing this?

```kotlin
    val deletedMeasurements = repository
        .deletedMeasurements(bloodPressure.uuid)
        .blockingFirst()
```

Reply:
This looks good actually. I'll switch to this.

Reply:
The actual problem with this test is that it is no longer essential for it to be an android test since  I moved the logic of filtering deleted blood pressures to the repository. I'll make this a JVM test instead and control the emissions myself to test it instead of working around Room's problems.

Reply:
Even better!

Reply:
Even better!